File: SolaceCore/Sources/SolaceCore/Models/SupportCircle.swift
import Foundation

/// A topic-based peer support group. Circles are curated/seeded, not
/// user-created — any signed-in student or counselor can browse, join, and
/// post.
public struct SupportCircle: Identifiable, Codable, Sendable, Equatable, Hashable {
    public let id: String
    public var name: String
    public var topicDescription: String
    public var memberIDs: [String]
    public var createdAt: Date

    public init(
        id: String,
        name: String,
        topicDescription: String,
        memberIDs: [String] = [],
        createdAt: Date = Date()
    ) {
        self.id = id
        self.name = name
        self.topicDescription = topicDescription
        self.memberIDs = memberIDs
        self.createdAt = createdAt
    }

    public func isMember(_ userID: String) -> Bool {
        memberIDs.contains(userID)
    }
}

File: SolaceAndroid/app/src/main/java/com/jonathanalumasa/solace/model/SupportCircle.kt
package com.jonathanalumasa.solace.model

import java.util.Date

/**
 * Mirrors `SolaceCore.SupportCircle`. Circles are curated/seeded, not
 * user-created — firestore.rules blocks client-side create/delete.
 */
data class SupportCircle(
    val id: String,
    val name: String,
    val topicDescription: String,
    val memberIDs: List<String> = emptyList(),
    val createdAt: Date = Date()
) {
    fun isMember(userID: String): Boolean = memberIDs.contains(userID)
}

/** Mirrors `SolaceCore.CircleMessage`. */
data class CircleMessage(
    val id: String,
    val circleID: String,
    val senderID: String,
    val senderName: String,
    val text: String,
    val sentAt: Date = Date()
)
